### PR TITLE
Change rocksmq retention config

### PR DESCRIPTION
**Pull request by xiaofan-luan**
related to #18304
1) Add LZ4 compression for rocksmq
2) trigger compaction when range delete message
3) change retention related parameters, retention size is changed to 2G, time is changed to 3days
Signed-off-by: xiaofan-luan <xiaofan.luan@zilliz.com>

### CODE REVIEW

**Comment by xiaofan-luan** [comment]
If we to test retention, we can change the time to 1 hour.
The way we want to keep some data in rocksmq in case some unexpected behaviour happened(For instance, cluster crashed for more than 3 days)

**Comment by xiaofan-luan** [comment]
@LoveEachDay pls help on testing the behaviour, thanks

**Comment by XuanYang-cn** [comment]
https://github.com/xiaofan-luan/milvus/pull/2 @xiaofan-luan

File: build/docker/builder/cpu/ubuntu18.04/Dockerfile [hunk 1]
```diff
@@ -14,10 +14,13 @@ FROM milvusdb/openblas:ubuntu18.04-20210428
 RUN apt-get update && apt-get install -y --no-install-recommends wget curl ca-certificates gnupg2 && \
     wget -qO- "https://cmake.org/files/v3.18/cmake-3.18.6-Linux-x86_64.tar.gz" | tar --strip-components=1 -xz -C /usr/local && \
     apt-get update && apt-get install -y --no-install-recommends \
-    g++ gcc gfortran git make ccache libssl-dev zlib1g-dev libboost-regex-dev libboost-program-options-dev libboost-system-dev libboost-filesystem-dev libboost-serialization-dev python3-dev libboost-python-dev libcurl4-openssl-dev libtbb-dev clang-format-10 clang-tidy-10 lcov libtool m4 autoconf automake && \
+    g++ gcc gfortran git make ccache libssl-dev zlib1g-dev libboost-regex-dev libboost-program-options-dev libboost-system-dev \
+    libboost-filesystem-dev libboost-serialization-dev python3-dev libboost-python-dev libcurl4-openssl-dev libtbb-dev clang-format-10 clang-tidy-10 lcov libtool m4 autoconf automake  && \
     apt-get remove --purge -y && \
     rm -rf /var/lib/apt/lists/*
 
+RUN apt-get update && apt-get install -y --no-install-recommends libzstd-devel
```
Comment (LoveEachDay) [inline build/docker/builder/cpu/ubuntu18.04/Dockerfile:22]: Should the package name be `libzstd-dev` for ubuntu?

Comment (XuanYang-cn) [reply build/docker/builder/cpu/ubuntu18.04/Dockerfile:22]: yes, and there're some other problems too. I fixed the compilation at https://github.com/xiaofan-luan/milvus/pull/2
@xiaofan-luan Please help review

**Comment by Sivannnnnn** [comment]
Hello, how is the current progress of this work? When is the version expected to be released? When I query frequently, the RDB file has been written all over my disk, and after I know the RDB file, I have lost some recent data

**Comment by xiaofan-luan** [comment]
@XuanYang-cn Thanks for the help~ you could open another pr with the fix and coauthor me, thought that would be a urgent issue

**Pull request closed without merging**